File: z-clients/javascript/src/models/kvDeleteIn.ts
// this file is @generated

export interface KvDeleteIn {
    namespace?: string | null;
    /**
     * If set, the delete only succeeds when the stored version matches this value.
     * Use the `version` field from a prior `get` response.
     */
    version?: number | null;
}

export interface KvDeleteIn_ {
    namespace?: string | null;
    key: string;
    /**
     * If set, the delete only succeeds when the stored version matches this value.
     * Use the `version` field from a prior `get` response.
     */
    version?: number | null;
}

export const KvDeleteInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): KvDeleteIn_ {
        return {
            namespace: object['namespace'],
            key: object['key'],
            version: object['version'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: KvDeleteIn_): any {
        return {
            'namespace': self.namespace,
            'key': self.key,
            'version': self.version,
        };
    }
}

File: z-clients/python/diom/models/kv_delete_in.py
# this file is @generated

from ..internal.base_model import BaseModel


class KvDeleteIn(BaseModel):
    namespace: str | None = None

    version: int | None = None
    """If set, the delete only succeeds when the stored version matches this value.
    Use the `version` field from a prior `get` response."""


class _KvDeleteIn(BaseModel):
    namespace: str | None = None

    key: str

    version: int | None = None
    """If set, the delete only succeeds when the stored version matches this value.
    Use the `version` field from a prior `get` response."""
